### PR TITLE
resolved_ts: fix `tracked_index` initialization

### PR DESCRIPTION
**Pull request by NingLin-P**
### What problem does this PR solve?

Problem Summary:

Related to https://github.com/tikv/tikv/pull/10095, the `tracked_index` didn't update to the snapshot's `apply_index`, if there are no locks in the snapshot and no incoming writes, the resolver's `tracked_index` will be 0 (which is wrong) and cause `safe_ts` can't update

### What is changed and how it works?

What's Changed:

Update the `tracked_index` to the snapshot's `apply_index` before tracking locks (if any) in the snapshot 

### Check List 

Tests 

- Integration test

### Release note 
- No release note

### CODE REVIEW

**Review by 5kbpers — APPROVED** [review]
rest LGTM

File: components/resolved_ts/src/endpoint.rs [hunk 1]
```diff
@@ -159,35 +159,39 @@ impl ObserveRegion {
...
-                                "pending data index" => tracked_index,
-                            );
-                        }
-                        ResolverStatus::Ready => {
-                            panic!("region {:?} resolver has ready", self.meta.id)
-                        }
-                    }
+                    // Update the `tracked_index` to the snapshot's `apply_index`
+                    self.resolver.update_tracked_index(apply_index);
+                    let tracked_index =
```
Comment (Yisaer) [inline components/resolved_ts/src/endpoint.rs:164]: Is the name `tracked_index` misleading here?

**Review by Yisaer — COMMENTED** [review]
Rest LGTM

**Review by Yisaer — APPROVED** [review]
/lgtm

**Pull request merged**